fv/wireguard_test.go

```diff
@@ -348,6 +335,52 @@ var _ = infrastructure.DatastoreDescribe("WireGuard-Supported", []apiconfig.Data
 			Expect(rcvd[1]).NotTo(BeEmpty())
 			Expect(rcvd[1]).To(Equal(sent[0]))
 		})
+
+		It("between pod to pod should be encrypted using wg tunnel with egress policy applied", func() {
+			policy := api.NewGlobalNetworkPolicy()
+
+			policy.Name = "f01-egress-deny"
+			policy.Spec.Egress = []api.Rule{{Action: api.Deny}}
+			policy.Spec.Selector = fmt.Sprintf("name in { '%s', '%s'}", wls[0].Name, wls[1].Name)
+			_, err := client.GlobalNetworkPolicies().Create(utils.Ctx, policy, utils.NoOptions)
+			Expect(err).NotTo(HaveOccurred())
+
+			cc.ExpectNone(wls[0], wls[1])
+			cc.ExpectNone(wls[1], wls[0])
+			cc.CheckConnectivity()
+
+			By("verifying tunnelled packet count is non-zero")
+			for i := range felixes {
+				Consistently(func() int {
+					return tcpdumps[i].MatchCount("numInTunnelPackets")
+				}, "5s", "100ms").Should(BeNumerically("==", 0))
+				Consistently(func() int {
+					return tcpdumps[i].MatchCount("numOutTunnelPackets")
+				}, "5s", "100ms").Should(BeNumerically("==", 0))
+			}
+
+			cc.ResetExpectations()
+
+			policy.Name = "f01-egress-allow"
+			policy.Spec.Egress = []api.Rule{{Action: api.Allow}}
+			policy.Spec.Selector = fmt.Sprintf("name in { '%s', '%s'}", wls[0].Name, wls[1].Name)
+			_, err = client.GlobalNetworkPolicies().Create(utils.Ctx, policy, utils.NoOptions)
+			Expect(err).NotTo(HaveOccurred())
+
+			cc.ExpectSome(wls[0], wls[1])
+			cc.ExpectSome(wls[1], wls[0])
+			cc.CheckConnectivity()
+
+			By("verifying tunnelled packet count is non-zero")
+			for i := range felixes {
+				Eventually(func() int {
+					return tcpdumps[i].MatchCount("numInTunnelPackets")
+				}, "10s", "100ms").Should(BeNumerically(">", 0))
+				Eventually(func() int {
+					return tcpdumps[i].MatchCount("numOutTunnelPackets")
+				}, "10s", "100ms").Should(BeNumerically(">", 0))
+			}
```

Comment:
I think 10s is gonna be too long for all of these tcp dump checks.  From a quick glance I think wireguard are the only tests that don't use the defaults (1s) for checking the tcp dump stats.  Do you think these need to be longer than the other tests?

Reply:
There are a few tests using `tcpdump` in available examples. Original motivation to bump from `5s` to `10s` was flakes seen earlier, which may have been addressed by recent changes. In terms of impact though, I think polling interval will have actual impact for `Eventually` than the total time which is actually from `10ms` to `100ms` which isn't huge IMO. We can revisit this if needed.